### PR TITLE
[RFC] Threat intelligence

### PR DESCRIPTION
**Pull request by shimonmodi**
Stage 0 submission for threat intelligence

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/ecs/blob/master/CONTRIBUTING.md)?
- For proposing substantial changes or additions to the schema, have you reviewed the [RFC process](https://github.com/elastic/ecs/blob/master/rfcs/README.md)?
- If submitting code/script changes, have you verified all tests pass locally using `make test`?
- If submitting schema/fields updates, have you generated new artifacts by running `make` and committed those changes?
- Is your pull request against master? Unless there is a good reason otherwise, we prefer pull requests against master and will backport as needed.
- Have you added an entry to the [CHANGELOG.next.md](https://github.com/elastic/ecs/blob/master/CHANGELOG.next.md)?

[Markdown preview of this RFC](https://github.com/shimonmodi/ecs/blob/shimonmodi-cti-1/rfcs/0000-threat-intel.md)

### CODE REVIEW

File: rfcs/0000-threat-intel.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+## References
+
+[Threat Intel Field Set Draft](https://docs.google.com/spreadsheets/d/1hS3tF-sGmwnKb7uUGLo3Rng_q6EFgwo6UCae8Sp4E-g/edit?usp=sharing)
+<!-- Insert any links appropriate to this RFC in this section. -->
+
+### RFC Pull Requests
+
+<!-- An RFC should link to the PRs for each of it stage advancements. -->
+
+* Stage 0: https://github.com/elastic/ecs/pull/NNN
```
Comment (ebeahan) [inline rfcs/0000-threat-intel.md:114]:
```suggestion
* Stage 0: https://github.com/elastic/ecs/pull/986
```

File: rfcs/0000-threat-intel.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+* @Monique | sponsor
+* @EunJung | subject matter expert
+* @JaneDoe | grammar, spelling, prose
+* @Mariana
+-->
+
+
+## References
+
+[Threat Intel Field Set Draft](https://docs.google.com/spreadsheets/d/1hS3tF-sGmwnKb7uUGLo3Rng_q6EFgwo6UCae8Sp4E-g/edit?usp=sharing)
```
Comment (ebeahan) [inline rfcs/0000-threat-intel.md:107]: If we include this spreadsheet as a reference, can we make it visible publicly for our community users?

File: rfcs/0000-threat-intel.md [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+<!--
+As you work on your RFC, use the "Stage N" comments to guide you in what you should focus on, for the stage you're targeting.
+Feel free to remove these comments as you go along.
+-->
+
+<!--
+Stage 0: Provide a high level summary of the premise of these changes. Briefly describe the nature, purpose, and impact of the changes. ~2-5 sentences.
+-->
+
+## Fields
```
Comment (ebeahan) [inline rfcs/0000-threat-intel.md:18]: Feel free to remove any unused section headers, and we can add them back in as the RFC is fleshed out in later stages.

File: rfcs/0000-threat-intel.md [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+<!-- Leave this ID at 0000. The ECS team will assign a unique, contiguous RFC number upon merging the initial stage of this RFC. -->
+
+- Stage: **0 (strawperson)** <!-- Update to reflect target stage. See https://elastic.github.io/ecs/stages.html -->
+- Date: **TBD** <!-- The ECS team sets this date at merge time. This is the date of the latest stage advancement. -->
+
+Elastic Security Solution will be adding the capability to ingest, process and utilize threat intelligence information for increasing detection coverage and helping analysts making quicker investigation decisions. Threat intelligence can be collected from a number of sources with a variety of structured and semi-structured data representations. This makes threat intelligence an ideal candidate for ECS mappings. Threat intelligence data will require ECS mappings to normalize it and make it usable in our security solution. This RFC is focused on identifying new field sets and values that need to be created for threat intelligence data. The goal is to utilize [STIX Cyber Observable data model](https://docs.oasis-open.org/cti/stix/v2.1/cs01/stix-v2.1-cs01.html#_mlbmudhl16lr) as a foundation for these new fields.
```
Comment (ebeahan) [inline rfcs/0000-threat-intel.md:7]: Looking over the draft spreadsheet, it looks like `threat.*` may also make use of nesting of other ECS fields (`file.*`, `url.*`, `as.*`, etc.).

Not necessary to get into specifics of which fields may nest for stage 0, but one possible addition here would be mention that field reuse is also being explored along with adding new fields and categorization values.

**Review by ebeahan — COMMENTED** [review]
Thanks for opening @shimonmodi!

I've added a few minor review comments, but I think overall this looks good to advance to stage 0.

**Comment by SHolzhauer** [comment]
See #1023 for a proposal on how to map IOC fields specifically.

File: rfcs/0000-threat-intel.md (hunk 2 above)
Comment (shimonmodi) [reply rfcs/0000-threat-intel.md:107]: yes - I can update the link to make it publicly viewable

File: rfcs/0000-threat-intel.md (hunk 4 above)
Comment (shimonmodi) [reply rfcs/0000-threat-intel.md:7]: agreed. I will add language to reflect that we will explore field reuse as a priority before adding new fields.

**Review by ebeahan — DISMISSED** [review]
Thanks @shimonmodi for making those changes! I'll assign the RFC # and merge the PR to advance to stage 0.

**Review by ebeahan — APPROVED** [review]

**Comment by webmat** [comment]
Thanks @ebeahan and @shimonmodi 👍 

Next step Shimon is simply to open a new PR for stage 1. You can start very quick by just changing "stage 0" to "stage 1" at the top of the doc, open the PR right away with only this change, and mark it as a draft.

Then you can work on the content of the stage 1 doc as time allows and push to that PR over time.

Opening the stage 1 quickly will give us a place to drop any further feedback and ideas on this, in the meantime.

**Comment by shimonmodi** [comment]
@webmat - thanks for the next steps. I just completed them.

**Comment by dainperkins** [comment]
Question on the excel sheet mappings - is it necessary to nest e.g. file, host, user, process, etc. under threat - or would it make more sense to use the top level fields in combination with the threat fields to make the overall document naming simpler?

**Comment by webmat** [comment]
@dainperkins Discussion about this should continue on the next stage PR, here https://github.com/elastic/ecs/pull/1037

**Pull request merged**